### ISSUE TITLE
missing support for SIGSEGV handler

### ISSUE THREAD

**Issue by stanislav-brabec** [Microsoft/BashOnWindows#1637]
* A brief description
WSL does not implement support for a custom SIGSEGV handler. As an example, it causes crash of lscpu from util-linux between 2.20 (the version where SIGSEGV handler was introduced) and 2.29.1 (at least).

Example of code that crashes:
https://git.kernel.org/cgit/utils/util-linux/util-linux.git/tree/sys-utils/lscpu.c?id=5bd2b14de42bd81abfee1a3f3265b2ecb92f79b3#n888

* Expected results
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigaction(SIGSEGV, {sa_handler=0x403ca0, sa_mask=[], sa_flags=SA_RESTORER|SA_SIGINFO, sa_restorer=0x7fabb294a610}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SI_KERNEL, si_addr=NULL} ---
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
=> The process continues after SIGSEGV.

* Actual results (with terminal output if applicable)
The process crashes
rt_sigaction(SIGSEGV, ) succeeds, but the SIGSEGV remains unhandled.

* Your Windows build number
14393.693

* Steps / All commands required to reproduce the error from a brand new installation 
Try lscpu
Output is empty, exit code is 1.

* Strace of the failing command
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigaction(SIGSEGV, {0x403a40, [], SA_RESTORER|SA_SIGINFO, 0x7f8ffde54950}, {SIG_DFL, [], SA_RESTORER, 0x7fcf159e4950}, 8) = 0
+++ exited with 1 +++
=> The process exits after SIGSEGV.

* Required packages and commands to install
util-linux between 2.20 and 2.29.1, tested on 2.28

**Comment by sunilmut** [comment]
@stanislav-brabec Looks like this may already be fixed in the Insider build [15002+] (https://msdn.microsoft.com/en-us/commandline/wsl/release_notes#build-15002). Adding @stehufntdev to confirm. You are on the Anniversary Update build (14393). These fixes will make it to the `Creators Update` release as well.

**Comment by stehufntdev** [comment]
I just tried lscpu version 2.27.1 on a recent internal build and it exits without printing anything.  From an strace, it is exiting cleanly due to missing sys\procfs files.

**Comment by stanislav-brabec** [comment]
I just prepared a patch for util-linux lscpu, so it should not crash any more, even with the old WSL version.

Feel free to comment the patch:
http://marc.info/?l=util-linux-ng&m=148579219928620&w=2

The output of lscpu will include two extra lines:

Hypervisor vendor:     Windows Subsystem for Linux
Virtualization type:   container

@stehufntdev Well, I only tested 2.28 and git master. Maybe other issues appear in earlier versions.
